sw/device/silicon_creator/rom/e2e/BUILD

```diff
@@ -1716,3 +1717,228 @@ test_suite(
         for b in SIGVERIFY_BAD_CASES
     ],
 )
+
+#################################################################
+# Adding some comments here to prevent interleaving during merges
+# from Git.
+# These lines will be deleted
+# TODO: remove these guard comments
+
+# Constrained to a specific device ID
+# TODO: what are some good test values here?
+_test_device_id = [
+    "ffffffff",  # Least-significant word
+    "12345678",
+    "00000003",
+    "abababab",
+    "cdcdcdcd",
+    "01010101",
+    "10101010",
+    "ffffffff",  # Most-significant word
+]
+
+_test_manuf_states = {
+    "creator": "fedcba98",
+    "owner": "12345678",
+}
+
+# Generate OTP image with a specific device ID, creator manufacturing state,
+# and owner manufacturing state
+otp_json(
+    name = "otp_json_set_device_id_overlay",
+    partitions = [
+        otp_partition(
+            name = "HW_CFG",
+            items = {"DEVICE_ID": "".join(_test_device_id[::-1])},
+        ),
+        otp_partition(
+            name = "CREATOR_SW_CFG",
+            items = {"CREATOR_SW_CFG_MANUF_STATE": _test_manuf_states["creator"]},
+        ),
+        otp_partition(
+            name = "OWNER_SW_CFG",
+            items = {"OWNER_SW_CFG_MANUF_STATE": _test_manuf_states["owner"]},
+        ),
+    ],
+    visibility = ["//visibility:private"],
+)
+
+otp_image(
+    name = "otp_img_set_device_id",
+    src = "//hw/ip/otp_ctrl/data:otp_json_rma",
+    overlays = [
+        "//hw/ip/otp_ctrl/data:otp_json_creator_sw_cfg",
+        "//hw/ip/otp_ctrl/data:otp_json_owner_sw_cfg",
+        "//hw/ip/otp_ctrl/data:otp_json_hw_cfg",
+        ":otp_json_set_device_id_overlay",
+    ],
+    visibility = ["//visibility:private"],
+)
+
+# TODO: is it okay to test everything with one bitstream?
+bitstream_splice(
```

Comment:
I currently have one bitstream that configures the device_id, creator manufacturing state, and owner manufacturing state. Is that okay, or should we test each of these in isolation (i.e. splice 3 separate bitstreams)?